### DISCUSSION TITLE
Lemmy support

### DISCUSSION THREAD

**Discussion by erlend-sh** [pfgithub/threadclient#9] [Ideas]
Right in between Mastodon and Reddit, there is [Lemmy](https://github.com/LemmyNet/lemmy) ;)

There’s also this project that provides a compatibility layer for Reddit clients to interoperate with Lemmy: https://github.com/derivator/tafkars

**Comment by pfgithub** [comment]
I was looking at lemmy but I couldn't find any documentation for their HTTP API. They do have a typescript client which I'll try

**Comment by pfgithub** [comment]
Also like mastodon there's no way of detecting if a link is to a lemmy page to keep it in-client. I'l need to figure something out for that. Mobile apps for mastodon often open an external web browser when you click a link to another mastodon post and that seems really bad to me, I don't want to have a client that does that

**Comment by pfgithub** [comment]
Unfortunately, it looks like the lemmy api is missing 'access-control-allow-origin' headers and threadclient would need to run a proxy server in order to communicate with lemmy instances. Lemmy support can't be added until this is changed: https://github.com/LemmyNet/lemmy/issues/3109

**Reply by erlend-sh** [reply]
Gotcha. Appreciate you looking into it and making an issue! I’ll keep an eye on it.
